.github/workflows/release_workspace.yml

```diff
@@ -1,56 +1,123 @@
 name: Release workspace
 
 on:
+  workflow_dispatch:
+    inputs:
+      workspace:
+        description: "Name of the Workspace"
+        required: true
+        type: string
   workflow_call:
     inputs:
       workspace:
+        description: "Name of the Workspace"
         required: true
         type: string
 
-defaults:
-  run:
-    working-directory: ./workspaces/${{ inputs.workspace }}
+concurrency:
+  group: ${{ github.workflow }}-${{ inputs.workspace }}
 
 jobs:
-  release:
-    timeout-minutes: 10
+  changesets-pr:
+    name: Update Version Packages PR for ${{ inputs.workspace }}
     runs-on: ubuntu-latest
-
+    defaults:
+      run:
+        working-directory: ./workspaces/${{ inputs.workspace }}
     env:
       CI: true
       NODE_OPTIONS: --max-old-space-size=4096
-
+    outputs:
+      publishing-needed: ${{ steps.should-publish.outputs.out }}
     steps:
       - name: Checkout
         uses: actions/checkout@v4
+      - name: Set up Node
+        uses: actions/setup-node@v4
         with:
-          ref: ${{ github.event.workflow_run.head_branch }}
+          node-version: 20
+          registry-url: https://registry.npmjs.org/ # Needed for auth
+
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "dir=$(yarn config get cacheFolder)" >> $GITHUB_OUTPUT
+
+      - uses: actions/cache@v3
+        id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles(format('workspaces/${0}/**/yarn.lock', inputs.workspace)) }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+      - name: yarn install
+        run: yarn install --immutable
 
+      - name: Detect changesets and eventually update Version Packages (${{ inputs.workspace }}) PR
+        id: changesets-pr
+        uses: changesets/action@v1
+        with:
+          title: Version Packages (${{ inputs.workspace }})
+          cwd: workspaces/${{ inputs.workspace }}
+          version: yarn changeset version
+        env:
+          # TODO(vinzscam): If we use secrets.GITHUB_TOKEN here, checks won't trigger in the version packages
+          # PR. cf. https://github.community/t/push-doesnt-trigger-workflow-action-in-an-open-pr/118675
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Early check whether publishing is needed
+        id: should-publish
+        if: steps.changesets-pr.outputs.hasChangesets == 'false'
```

Comment:
I don't think I follow this one perfectly 🤔 If there were NO changesets - we run the script. But if there were changesets we definitely don't release at all? Are you using the absence of changeset files as a sign that a merge was just made that removed all of those files which probably was a Version Packages merge?

Reply:
Yes, if there are changesets we just update the Version packages PR and don't release at all. I'm following the same logic as `changesets/action`: https://github.com/changesets/action/blob/main/src/index.ts#L105